examples/infection-by-group/README.md

```diff
@@ -0,0 +1,380 @@
+        let end_county_index = context.sample_range(MovementRng, 0..k);
+        // get the associated county
+        let county_iter = Counties::iter();
+        end_county = county_iter.next();
+        for i in 0..end_county_index {
+            end_county = county_iter.next();
+        }
+        // get the movement rate
+        let movement_rate = movement_rates.get(end_county);
+    }
+    context.add_plan(move |context|
+                     // i conceptualize ixa as knowing that Counties is one-to-one, so
+                     // this "assignment" is actually a "remove" person from last group
+                     // and then assign them to this new group
+                     // in other words, always staying in line with the group mapping requirements
+                     {context.assign_person_to_group(person_id, end_county);
+                      move_person(person_id);},
+                      context.get_time() + context.sample_distr(MovementRng,
+                                                                Exp::new(movement_rate).unwrap()));
+}
+```
+
+### Scheduling infection based on time-varying foi
+
+In `transmission.rs`:
+
+```rust
+use roots::find_root_brent;
+use reikna::integral::*;
+define_rng!(InfectionRng);
+
+let infection_plans: HashMap<PersonID, Id> = HashMap::new()
+
+fn init(context: &mut Context) {
+    context.subscribe_to_event(PersonCreationEvent, expose_person_to_deviled_eggs);
+    context.subscribe_to_event(PersonGroupUpdateEvent<Counties>, reexpose_person_to_deviled_eggs);
+}
+
+fn expose_person_to_deviled_eggs(context: &mut Context,
+                                 person_creation_event: PersonCreationEvent) {
+    inverse_sampling_infection(context, person_creation_event.person_id());
+}
+
+fn reexpose_person_to_deviled_eggs(context: &mut Context,
+                                   person_group_update_event: PersonGroupUpdateEvent<Counties>) {
+    inverse_sampling_infection(context, person_group_update_event.person_id());
+}
+
+// parameterize the foi
+fn foi(t: f64, sin_shift: f64) -> f64 {
+    f64::sin(t + sin_shift) + 1
+}
+
+// Evaluate whether infection can happen analytically -- because we know foi(t) exactly
+// here's how the method works: foi(t) describes the hazard rate for infection at a
+// given time, so 1 - e ^ -\int_0^t foi(u)du is the cumulative probability of infection at t.
+// Therefore, infection has happened by time t with the probability described above.
+// Drawing a random number on U(0, 1) gives a probability for the infection happening when the
+// CDF hits that value. To find the time at which this occurs, solve for the value of t
+// that gives that value of the CDF.
+// The math behind this is the same as drawing s ~ Exp(1), setting that equal to \int_0^t foi(u)du,
+// and solving for the time t at which the infection happens, which is fewer steps/easier.
+fn inverse_sampling_infection(context: &mut Context, person_id: PersonID) {
```

Comment:
This is nice as a contrast to the other basic infection model. I wonder if this actually ends up being its own mini example?

Reply:
This is now in #54.